tracy-edit/src/tracy-edit.cpp

```diff
@@ -0,0 +1,213 @@
+#ifdef _WIN32
+#  include <windows.h>
+#endif
+
+#include <fstream>
+#include <iostream>
+#include <regex>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "../../server/TracyFileRead.hpp"
+#include "../../server/TracyFileWrite.hpp"
+#include "../../server/TracyWorker.hpp"
+#include "../../getopt/getopt.h"
+
+#include "OfflineSymbolResolver.h"
+
+struct Args 
+{
+    const char* inputTracyPath = nullptr;
+    const char* outputTracyPath = nullptr;
+    bool verbose = false;
+    bool resolveSymbols = false;
+    std::vector<std::string> pathSubstitutions;
+    tracy::FileWrite::Compression compressionType = tracy::FileWrite::Compression::Zstd;
+    int compressionLevel = 5;
+};
+
+void PrintUsageAndExit()
+{
+    std::cerr << "Modify a tracy file" << std::endl;
+    std::cerr << "Usage:" << std::endl;
+    std::cerr << "  tracy-edit [OPTION...] <input trace file> <output tracy file>" << std::endl;
+    std::cerr << std::endl;
+    std::cerr << "  -h, --help                               Print usage" << std::endl;
+    std::cerr << "  -v, --verbose                            Enable verbose logging" << std::endl;
+    std::cerr << "  -r, --resolveSymbols                     Resolve symbols and patch callstack frames" << std::endl;
+    std::cerr << "  -s, --substitutePath \"REGEX_MATCH;REPLACEMENT\" Substitute symbol resolution path with an alternative" << std::endl;
+    std::cerr << "  -c, --compression Type                   Set the compression algorithm used for the output [Fast,Slow,Extreme,Zstd]" << std::endl;
+    std::cerr << "  -l, --compressesionLevel Level           Level of compression" << std::endl;
+    exit( 1 );
+}
+
+static const char* compressionTypeStr[]
+{
+    "Fast",
+    "Slow",
+    "Extreme",
+    "Zstd"
+};
+static_assert( uint32_t(tracy::FileWrite::Compression::Zstd)+1 == sizeof(compressionTypeStr)/sizeof(compressionTypeStr[0]));
+tracy::FileWrite::Compression getCompressionFromString(const char* str)
+{
+    for( uint32_t i = 0; i < sizeof(compressionTypeStr)/sizeof(compressionTypeStr[0]); ++i )
+    {
+        if( strcmp( compressionTypeStr[i], str ) == 0 )
+        {
+            return tracy::FileWrite::Compression( i );
+        }
+    }
+    return tracy::FileWrite::Compression::Zstd;
+}
+
+Args ParseArgs( int argc, char** argv )
+{
+    if ( argc < 3 )
+    {
+        PrintUsageAndExit();
+    }
+
+    Args args;
+
+    struct option long_opts[] = 
+    {
+        { "help", no_argument, NULL, 'h' },
+        { "verbose", no_argument, NULL, 'v' },
+        { "resolveSymbols", no_argument, NULL, 'r' },
+        { "substitutePath", required_argument, NULL, 's' },
+        { "compression", required_argument, NULL, 'c' },
+        { "compressesionLevel", required_argument, NULL, 'l' },
+        { NULL, 0, NULL, 0 }
+    };
```

Comment:
Can you unify processing of compression parameters with the `update` utility?

Reply:
oh, I have to admit I didn't realize the update tool already did that. Maybe I should actually move the symbol resolution to it instead of creating the tracy-edit tool then?

Reply:
It may make sense to do that. Try to keep the new functionality in a separate source file.

Reply:
done